src/napari_imagej/widgets/napari_imagej.py

```diff
@@ -153,6 +155,16 @@ def _handle_output(self, data):
         else:
             raise TypeError(f"Do not know how to display {data}")
 
+    @Slot(object)
+    def _update_progress(self, event: "jc.ModuleEvent"):
+        module = event.getModule()
+        if isinstance(event, jc.ModuleExecutingEvent):
+            _update_progress(module)
+        if isinstance(event, jc.ModuleExecutedEvent):
+            _update_progress(module)
+        elif isinstance(event, jc.ModuleFinishedEvent):
+            _update_progress(module)
```

Comment:
Does this need to be done on the GUI thread? Or could we just put this within the `EventListener`?